src/neptune/internal/utils/dependency_tracking.py

```diff
@@ -0,0 +1,47 @@
+__all__ = [
+    "DependencyTrackingStrategy",
+    "InferDependenciesStrategy",
+    "FileDependenciesStrategy",
+]
+
+import os
+import subprocess
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import Union
+
+from neptune.types import File
+
+
+class DependencyTrackingStrategy(ABC):
+    @abstractmethod
+    def track_dependencies(self) -> None:
+        ...
+
+
+class InferDependenciesStrategy(DependencyTrackingStrategy):
+    def __init__(self, run):
```

Comment:
Type hints?

Reply:
+ `run` is common for both strategies, maybe move it to parameter of `track_dependencies`?

Reply:
Added with the TYPE_CHECKING pattern to avoid circular imports